### PR TITLE
Fairphone 2 support

### PR DESCRIPTION
**Pull request by z3ntu**
* `while [ ! ]; do cat /sys/class/graphics/fb0/modes > /sys/class/graphics/fb0/mode; done` is needed for screen output
* Basically nothing else was tested

img_20170920_135757

### CODE REVIEW

**Comment by z3ntu** [comment]
I just saw `msm-fb-refresher` so I'll test it with that flag enabled.

**Comment by drebrez** [comment]
@z3ntu can you please try also without the `ENV{WL_CALIBRATION}=...` line in the udev rule? the values are close to the default any maybe it works without the need to calibrate.
Do you also have a `profile.sh` which starts only few apps, did you try adding `deviceinfo_weston_core_modules="xwayland.so"` into the deviceinfo and removing that profile.sh script to see if `postmarketos-demos` app starts?

**Comment by ollieparanoid** [comment]
Touch will probably work, as soon as you specify the right touch device name in the weston rule. See [Weston screen calibration](https://wiki.postmarketos.org/wiki/Screen-Calibration#Weston) for more information (@drebrez: just adjusted that article, so it doesn't suggest using `ENV{WL_CALIBRATION}` when not needed anymore).

@z3ntu, thanks for making the PR!

*Added the `dont_merge_pr` label according to the comments above, please tell us when it's ready to be reviewed!*

File: aports/device/device-fairphone-fp2/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
+pkgname=device-fairphone-fp2
+pkgver=1
+pkgrel=1
+pkgdesc="Fairphone 2"
+url="https://github.com/postmarketOS"
+arch="noarch"
+license="MIT"
+depends="linux-fairphone-fp2 linux-firmware mkbootimg"
```
[the remark this answers is not in the document]
Comment (ollieparanoid) [reply aports/device/device-fairphone-fp2/APKBUILD]: yes

**Comment by z3ntu** [comment]
I've run the calibrator 5 times

```
localhost:~# weston-calibrator 
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
Calibration values: -1.044899 0.558739 444.873474 -1.268061 -0.471630 2385.155762
localhost:~# weston-calibrator 
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
Calibration values: -0.763826 0.779871 -16.082214 -1.025998 -0.174601 1843.401001
localhost:~# weston-calibrator 
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
Calibration values: -0.822636 0.801112 5.112671 -1.039684 -0.212790 1889.703491
localhost:~# weston-calibrator 
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
Calibration values: -0.784515 0.774140 6.552246 -1.019672 -0.202881 1869.272095
localhost:~# weston-calibrator 
could not load cursor 'dnd-move'
could not load cursor 'dnd-copy'
could not load cursor 'dnd-none'
Calibration values: -0.782875 0.774699 17.572754 -1.084715 -0.179877 1877.924927
```
Touch screen device name is `synaptics_dsx` and with that in the udev rule, touchscreen works.

With `deviceinfo_msm_refresher="true"` the splashscreen shows with "Loading..." but after a couple of seconds the phone reboots. After commenting that flag again, pmOS works again, but without the screen...

**Comment by ollieparanoid** [comment]
Have you added a dependency on `msm-fb-refresher` in the device package (we *should* check that, but just making sure)?

**Comment by ollieparanoid** [comment]
Then it is a bug in either the kernel drivers, `msm-fb-refresher` or in the way we use it. Could you open another issue for that? I suggest leaving it out of this PR then.

**Comment by z3ntu** [comment]
@drebrez With the xwayland.so added, postmarketos-demos starts and it lets me start other apps. So xwayland seems to work fine.

**Comment by z3ntu** [comment]
Created an issue for msm-fb-refresher: #604

**Comment by z3ntu** [comment]
Splash screen: no
Weston: only with the beautiful while loop

**Comment by MartijnBraam** [comment]
Maybe the msm refresher needs a patch to set the mode instead of the panning. I guess the panning works on mdp3/mdp4 devices and mdss needs to set the mode every frame

**Pull request merged**